borsh-derive-internal/src/attribute_helpers.rs

```diff
@@ -18,3 +29,161 @@ pub fn contains_initialize_with(attrs: &[Attribute]) -> Option<Path> {
 
     None
 }
+
+pub fn check_item_attributes(derive_input: &DeriveInput) -> Result<(), proc_macro2::TokenStream> {
+    for attr in derive_input.attrs.clone() {
+        if attr.path().is_ident(BORSH_TOP) {
+            if let syn::Data::Struct(ref _data) = derive_input.data {
+                return Err(TokenStream::from(
+                    syn::Error::new(
+                        derive_input.ident.span(),
+                        "borsh (use_discriminant=<bool>) does not support structs",
+                    )
+                    .to_compile_error(),
+                ));
+            }
+        }
+    }
+    Ok(())
+}
+
+pub fn contains_use_discriminant(
+    attrs: &[Attribute],
+) -> Result<Option<bool>, proc_macro2::TokenStream> {
+    for attr in attrs {
+        if attr.path().is_ident(BORSH_TOP) {
+            let span = attr.path().get_ident().unwrap().span();
+            if let Meta::List(list) = attr.meta.clone() {
```

Comment:
this code actually passes 
```rust
#[derive(BorshDeserialize)]
#[borsh(foo)]
struct E1<'a, 'b> {
    a: &'a A<'b>,
}
```
with no error like
```bash
 1  error: `use_discriminant` is the only supported attribute for `borsh`
```
See example below with `attr.parse_nested_meta` in `check_item_attributes(...)`, which does error in this case